vllm/core/block_manager.py

```diff
@@ -295,8 +387,9 @@ def append_slot(
             # Not shared with other sequences. Appendable.
             # If the last block is now complete, promote it to a full block so
             # that it can be shared
-            new_block = self._maybe_promote_last_block(seq, last_block)
-            block_table[-1] = new_block
+            if self.enable_caching:
+                new_block = self._maybe_promote_last_block(seq, last_block)
+                block_table[-1] = new_block
```

Comment:
Make things a bit more clear. 
```suggestion
            if self.enable_caching:
                # If the last block is now complete, we may reuse an old block to save memory
                maybe_new_block = self._maybe_promote_last_block(seq, last_block)
                block_table[-1] = maybe_new_block
```
Please remove the two line comments above:
```
            # If the last block is now complete, promote it to a full block so
            # that it can be shared
```